In [1]:
import numpy as np
import pandas as pd

In [2]:
dataset = pd.read_csv("car_data.csv")

In [3]:
dataset.head()

Car_Name  Year  Selling_Price  Present_Price  Kms_Driven Fuel_Type  \
0     ritz  2014           3.35           5.59       27000    Petrol   
1      sx4  2013           4.75           9.54       43000    Diesel   
2     ciaz  2017           7.25           9.85        6900    Petrol   
3  wagon r  2011           2.85           4.15        5200    Petrol   
4    swift  2014           4.60           6.87       42450    Diesel   

  Seller_Type Transmission  Owner  
0      Dealer       Manual      0  
1      Dealer       Manual      0  
2      Dealer       Manual      0  
3      Dealer       Manual      0  
4      Dealer       Manual      0

<h4>Encoding</h4> 

In [4]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
dataset["Fuel_Type"] = le.fit_transform(dataset["Fuel_Type"])
dataset["Seller_Type"] = le.fit_transform(dataset["Seller_Type"])
dataset["Transmission"] = le.fit_transform(dataset["Transmission"])

<h4>Droping Car_Name Column</h4>

In [7]:
dataset = dataset.drop(columns ="Car_Name",axis = 1)

In [8]:
dataset.isnull().sum()

Year             0
Selling_Price    0
Present_Price    0
Kms_Driven       0
Fuel_Type        0
Seller_Type      0
Transmission     0
Owner            0
dtype: int64

In [9]:
x = dataset.drop(columns ="Owner",axis = 1)
x.head()

Year  Selling_Price  Present_Price  Kms_Driven  Fuel_Type  Seller_Type  \
0  2014           3.35           5.59       27000          2            0   
1  2013           4.75           9.54       43000          1            0   
2  2017           7.25           9.85        6900          2            0   
3  2011           2.85           4.15        5200          2            0   
4  2014           4.60           6.87       42450          1            0   

   Transmission  
0             1  
1             1  
2             1  
3             1  
4             1

In [10]:
y = dataset["Owner"]

In [11]:
from sklearn.model_selection import train_test_split


In [12]:
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size = 0.3,random_state = 90)

In [13]:
x_train.shape,x_test.shape,y_train.shape,y_test.shape

((210, 7), (91, 7), (210,), (91,))

In [14]:
x_train.iloc[:,4:].head().values
x_train[["Fuel_Type","Seller_Type","Transmission"]]

Fuel_Type  Seller_Type  Transmission
60           2            0             1
238          2            0             1
57           2            0             1
262          2            0             1
253          1            0             1
..         ...          ...           ...
58           2            0             1
167          2            1             1
159          2            1             0
114          2            1             1
157          2            1             1

[210 rows x 3 columns]

In [15]:
import tensorflow as tf
from tensorflow import keras

In [16]:
model= keras.Sequential([
    keras.layers.Dense(7,input_dim = 7,activation = 'relu'),
    keras.layers.Dense(14,activation = 'relu'),
    keras.layers.Dense(14,activation = 'relu'),
    keras.layers.Dense(7,activation = 'relu'),
    keras.layers.Dense(1,activation = 'sigmoid')
])
model.compile(loss = 'binary_crossentropy',optimizer = 'adam',metrics = ['accuracy'])
model.fit(x_train,y_train,epochs = 200,batch_size = 2)

Epoch 1/200
105/105 [==============================] - 0s 913us/step - loss: 61.9084 - accuracy: 0.9524
Epoch 2/200
105/105 [==============================] - 0s 904us/step - loss: 11.4303 - accuracy: 0.9143
Epoch 3/200
105/105 [==============================] - 0s 923us/step - loss: 17.1932 - accuracy: 0.9381
Epoch 4/200
105/105 [==============================] - 0s 913us/step - loss: 8.0380 - accuracy: 0.9190
Epoch 5/200
105/105 [==============================] - 0s 904us/step - loss: 10.3696 - accuracy: 0.9429
Epoch 6/200
105/105 [==============================] - 0s 923us/step - loss: 34.5619 - accuracy: 0.9571
Epoch 7/200
105/105 [==============================] - 0s 913us/step - loss: 14.0102 - accuracy: 0.9476
Epoch 8/200
105/105 [==============================] - 0s 913us/step - loss: 2.9917 - accuracy: 0.9190
Epoch 9/200
105/105 [==============================] - 0s 913us/step - loss: 7.7595 - accuracy: 0.9429
Epoch 10/200
105/105 [==============================] - 0s 904us/st

105/105 [==============================] - 0s 885us/step - loss: 0.3232 - accuracy: 0.9476
Epoch 80/200
105/105 [==============================] - 0s 904us/step - loss: 0.5807 - accuracy: 0.9429
Epoch 81/200
105/105 [==============================] - 0s 894us/step - loss: 0.5710 - accuracy: 0.9381
Epoch 82/200
105/105 [==============================] - 0s 894us/step - loss: 0.2679 - accuracy: 0.9476
Epoch 83/200
105/105 [==============================] - 0s 885us/step - loss: 0.3919 - accuracy: 0.9476
Epoch 84/200
105/105 [==============================] - 0s 894us/step - loss: 0.4398 - accuracy: 0.9381
Epoch 85/200
105/105 [==============================] - 0s 894us/step - loss: 0.6518 - accuracy: 0.9381
Epoch 86/200
105/105 [==============================] - 0s 942us/step - loss: 0.3885 - accuracy: 0.9476
Epoch 87/200
105/105 [==============================] - 0s 933us/step - loss: 0.3018 - accuracy: 0.9476
Epoch 88/200
105/105 [==============================] - 0s 933us/step - loss:

105/105 [==============================] - 0s 856us/step - loss: 0.1912 - accuracy: 0.9571
Epoch 158/200
105/105 [==============================] - 0s 865us/step - loss: 0.1885 - accuracy: 0.9571
Epoch 159/200
105/105 [==============================] - 0s 856us/step - loss: 0.1909 - accuracy: 0.9571
Epoch 160/200
105/105 [==============================] - 0s 913us/step - loss: 0.1991 - accuracy: 0.9571
Epoch 161/200
105/105 [==============================] - 0s 894us/step - loss: 0.1878 - accuracy: 0.9571
Epoch 162/200
105/105 [==============================] - 0s 894us/step - loss: 0.1948 - accuracy: 0.9571
Epoch 163/200
105/105 [==============================] - 0s 894us/step - loss: 0.1770 - accuracy: 0.9571
Epoch 164/200
105/105 [==============================] - 0s 1ms/step - loss: 0.1830 - accuracy: 0.9571
Epoch 165/200
105/105 [==============================] - 0s 923us/step - loss: 0.2122 - accuracy: 0.9571
Epoch 166/200
105/105 [==============================] - 0s 904us/step 

accuracy  = 95%

In [17]:
from tensorflow.keras.models import save_model
save_model(model, "model.h5")

# load and evaluate a saved model
from numpy import loadtxt
from tensorflow.keras.models import load_model
 
# load model
model = load_model('model.h5')
# summarize model.
model.summary()
# load dataset
dataset = loadtxt("pima-indians-diabetes.csv", delimiter=",")
# split into input (X) and output (Y) variables
X = dataset[:,0:8]
Y = dataset[:,8]
# evaluate the model
score = model.evaluate(X, Y, verbose=0)
print("%s: %.2f%%" % (model.metrics_names[1], score[1]*100))

Model: "sequential_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 dense_5 (Dense)             (None, 7)                 56   
 
                                                                 
 dense_6 (Dense)             (None, 14)                112  
 
                                                                 
 dense_7 (Dense)             (None, 14)                210 
 
                                                                 
 dense_8 (Dense)             (None, 7)                 105 
 
                                                                 
 dense_9 (Dense)             (None, 1)                 8  
 
                                                                 
=================================================================

Total params: 491

Trainable params: 491

Non-trainable params: 0

_________________________________________________________________


from numpy import loadtxt
from tensorflow.keras.models import load_model
model = load_model('model.h5')

In [18]:
from numpy import loadtxt
from tensorflow.keras.models import load_model
model = load_model('model.h5')
model.predict(x_test[:1])


1/1 [==============================] - 0s 85ms/step


array([[0.05534448]], dtype=float32)